phpunit/class-block-templates-test.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * REST API: Block_Templates_Test class
+ *
+ * @package    WordPress
+ * @subpackage REST_API
+ */
+
+/**
+ * Tests for the Block Templates abstraction layer.
+ */
+class Block_Templates_Test extends WP_UnitTestCase {
+	private static $post;
+
+	public static function wpSetUpBeforeClass() {
+		switch_theme( 'tt1-blocks' );
+		gutenberg_register_template_post_type();
+		gutenberg_register_template_part_post_type();
+		gutenberg_register_wp_theme_taxonomy();
+
+		$args = array(
+			'post_type'    => 'wp_template',
+			'post_name'    => 'my_template',
+			'post_title'   => 'My Template',
+			'post_content' => 'Content',
+			'post_excerpt' => 'Description of my template',
+			'tax_input'    => array(
+				'wp_theme' => array(
+					get_stylesheet(),
+				),
+			),
+		);
+
+		self::assertTrue( taxonomy_exists( 'wp_theme' ) );
+		self::$post = self::factory()->post->create_and_get( $args );
+		wp_set_post_terms( self::$post->ID, get_stylesheet(), 'wp_theme' );
+	}
+
+	public static function wpTearDownAfterClass() {
+		wp_delete_post( self::$post->ID );
+	}
+
```

Comment:
```suggestion
```

Not needed, the posts table is reset after the class tearDown. It's a lot quicker too as the DB is simply rolled back.

Reply:
That is only for `tearDown` though, no? If the fixture is created in `setUpBeforeClass` the transaction is committed. And the transaction isn't started until the end of `setUp`.

Reply:
So this changed in https://core.trac.wordpress.org/changeset/38398. Now we have a `_delete_all_data` call to account for the things that the transactions can't handle. I know at least in the REST API tests we follow the delete in `wpTearDownAfterClass` pattern, I'll work on dropping those.